nodeInterface/NodeInterface.go

```diff
@@ -461,6 +447,68 @@ func (n NodeInterface) GasEstimateComponents(
 	if !contractCreation {
 		args.To = &to
 	}
+	return args
+}
+
+func (n NodeInterface) GasEstimateL1Component(
+	c ctx, evm mech, value huge, to addr, contractCreation bool, data []byte,
+) (uint64, huge, huge, error) {
+	node, err := arbNodeFromNodeInterfaceBackend(n.backend)
+	if err != nil {
+		return 0, nil, nil, err
+	}
+	if to == types.NodeInterfaceAddress || to == types.NodeInterfaceDebugAddress {
+		return 0, nil, nil, errors.New("cannot estimate virtual contract")
+	}
+
+	backend := node.Backend.APIBackend()
+	gasCap := backend.RPCGasCap()
+	args := n.messageArgs(evm, value, to, contractCreation, data)
+
+	// assign 0 so that PosterDataCost uses its better value
+	zeroGas := hexutil.Uint64(0)
+	args.Gas = &zeroGas
```

Comment:
I don't think this works because ToMessage will override it

Reply:
Nice catch, I've changed it to use the random value from `l1pricing` directly